### PR TITLE
Add Static Variable Clone

### PR DESCRIPTION
**Pull request by zhhsplendid**
### PR types

New features

### PR changes

APIs

### Describe

Add clone method for static Variable so that this interface will be same as dygraph. It fixed some bugs in dy2stat

### CODE REVIEW

**Review by liym27 — DISMISSED** [review]
LGTM

**Review by chalsliu — DISMISSED** [review]

File: python/paddle/fluid/framework.py [hunk 1]
```diff
@@ -1629,6 +1629,41 @@ def type(self):
...
+                y = x.clone()
+
+        """
+        main_program = default_main_program()
+        output = main_program.current_block().create_var(
+            name=unique_name.generate_with_ignorable_key(self.name + "_clone"),
+            dtype=self.dtype,
+            type=self.type,
+            persistable=self.persistable,
+            stop_gradient=self.gradient)
```
Comment (zhiqiu) [inline python/paddle/fluid/framework.py]:
```suggestion
            stop_gradient=self.stop_gradient)
```

File: python/paddle/fluid/framework.py [hunk 2]
```diff
@@ -1629,6 +1629,41 @@ def type(self):
...
+                paddle.enable_static()
+
+                # create a static Variable
+                x = paddle.static.data(name='x', shape=[3, 2, 1])
+                # create a cloned Variable
+                y = x.clone()
+
+        """
+        main_program = default_main_program()
+        output = main_program.current_block().create_var(
```
Comment (zhiqiu) [inline python/paddle/fluid/framework.py]: Better `self.block.create_var(..)` and `apend_op`

Comment (zhhsplendid) [reply python/paddle/fluid/framework.py]: Done

File: python/paddle/fluid/framework.py (hunk 1 above)
Comment (zhhsplendid) [reply python/paddle/fluid/framework.py]: Done

**Review by zhiqiu — DISMISSED** [review]
LGTM for Variable.clone

**Review by liym27 — APPROVED** [review]

**Review by chalsliu — APPROVED** [review]

**Review by swtkiwi — APPROVED** [review]
LGTM for framework.py

**Review by Xreki — APPROVED** [review]
LGTM for the modification of framework.py

**Review by zhiqiu — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**